deeppavlov/models/multitask_pal_bert/multitask_pal_bert.py

```diff
@@ -323,19 +322,21 @@ def train_on_batch(self, *args):
 
         if self.tasks_type[task_id] == "regression":
             _input["labels"] = torch.tensor(
-                task_labels, dtype=torch.float32
-            ).to(self.device)
+                task_labels, dtype=torch.float32).to(self.device)
         else:
-            _input["labels"] = torch.from_numpy(np.array(task_labels)).to(
-                self.device
-            )
-
+            _input["labels"] = torch.tensor(
+                task_labels, dtype=torch.long).to(self.device)
         loss, logits = self.model(
             task_id=task_id, name=self.tasks_type[task_id], **_input
         )
 
         loss = loss / self.gradient_accumulation_steps
-        loss.backward()
+        try:
+            loss.backward()
+        except RuntimeError:
+            raise ValueError(
```

Comment:
In that case, the problem is not there, it is in the forward function of BertForMultiTask, loss_fst calculation

Reply:
I think we can catch the same here as well, as it does throw an error while calling `loss.backward()` due to this mismatch. 
I did not catch it in modeling.py because there in the forward function I only have access to task_id, not task names. So I cannot raise an error with the task name.